paddle/framework/tensor.h

```diff
@@ -0,0 +1,91 @@
+/*
+  Copyright (c) 2016 PaddlePaddle Authors. All Rights Reserve.
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+  http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+#pragma once
+
+namespace paddle {
+namespace framework {
+
+class Tensor {
+  using paddle::platform::Place;
+  using paddle::platform::get_place;
+
+ public:
+  explicit Tensor(DDim dims) : dims_(dims), place_(get_place()) {}
+  explicit Tensor(DDim dims, Place place) : dims_(dims), place_(place) {}
+
+  template <typename T>
+  const T* data() const {
```

Comment:
Cite from the link you pasted:

> Accessors and mutators (get and set functions) may be named like variables. These often correspond to actual member variables, but this is not required. For example, int count() and void set_count(int count).

Here we use `data` and `mutable_data` since they are accessors of Tensor.